From 9e7ba8e73d8e043a91d215af4b4fcdb8e37921f5 Mon Sep 17 00:00:00 2001
From: Michael Marineau <mike@marineau.org>
Date: Tue, 27 Oct 2020 19:47:20 +0100
Subject: [PATCH 2/3] environment: Filter EROOT for all EAPIs

This variable is often defined in older EAPIs with "${EROOT:=$ROOT}"
but it should never be preserved since ROOT may change. Bug #490014

https://bugs.gentoo.org/490014
---
 bin/phase-functions.sh | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/bin/phase-functions.sh b/bin/phase-functions.sh
index 90e622e..ec7b18b 100644
--- a/bin/phase-functions.sh
+++ b/bin/phase-functions.sh
@@ -107,10 +107,14 @@ __filter_readonly_variables() {
 	if ___eapi_has_BROOT; then
 		filtered_vars+=" BROOT"
 	fi
-	# Don't filter/interfere with prefix variables unless they are
-	# supported by the current EAPI.
+
+	# Always filter EROOT to ensure it is redefined based on ROOT
+	filtered_vars+=" EROOT"
+
+	# Don't filter/interfere with the other prefix variables unless
+	# they are supported by the current EAPI.
 	if ___eapi_has_prefix_variables; then
-		filtered_vars+=" ED EPREFIX EROOT"
+		filtered_vars+=" ED EPREFIX"
 		if ___eapi_has_SYSROOT; then
 			filtered_vars+=" ESYSROOT"
 		fi
-- 
2.26.2

